### PR TITLE
Evaluate eeitt context variables from the eeitt service

### PR DESCRIPTION
**Pull request by lutnos**
Evaluate eeitt context variables from the eeitt service for customeId at declaration

### CODE REVIEW

File: app/uk/gov/hmrc/gform/auditing/AuditService.scala [hunk 1]
```diff
@@ -64,26 +64,26 @@ trait AuditService {
     processedData.map(x => x.id.value -> x.value).toMap
   }
 
-  def sendSubmissionEvent(form: Form, sections: List[BaseSection], retrievals: Retrievals)(implicit ec: ExecutionContext, hc: HeaderCarrier, request: Request[_]) = {
-    sendEvent(form, formToMap(form, sections), retrievals)
+  def sendSubmissionEvent(form: Form, sections: List[BaseSection], retrievals: Retrievals, customerId: String)(implicit ec: ExecutionContext, hc: HeaderCarrier, request: Request[_]) = {
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/auditing/AuditService.scala]: This `customerId` is great candidate for lifting into value class, so we avoid working with `String` type. I'd be happy to see this change, but it does not necessary needs to be part of this PR.

File: app/uk/gov/hmrc/gform/auth/AuthService.scala [hunk 2]
```diff
@@ -32,17 +35,19 @@ class AuthService {
       .fold("")(_.getIdentifier(EEITTAuthConfig.nonAgentIdName).fold("")(_.value))
   }
 
-  def evaluateSubmissionReference(expression: TextExpression, retrievals: Retrievals, data: Map[FormComponentId, Seq[String]]): String = {
+  def evaluateSubmissionReference(expression: TextExpression, retrievals: Retrievals, formTemplate: FormTemplate,
+    data: Map[FormComponentId, Seq[String]])(implicit hc: HeaderCarrier): Future[String] = {
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/auth/AuthService.scala]: So this method returns `customerId`, but it is really hard to see. Once we would introduced `CustomerId` value class, return method `Future[CustomerId]` would be more pleasant to read and understand.

File: app/uk/gov/hmrc/gform/gform/AcknowledgementController.scala [hunk 3]
```diff
@@ -87,49 +88,52 @@ class AcknowledgementController(
     submissionReference: Option[TextExpression],
     retrievals: Retrievals,
     hashedValue: String,
+    formTemplate: FormTemplate,
     data: Map[FormComponentId, Seq[String]]
-  ): String = {
+  )(implicit hc: HeaderCarrier): Future[String] = {
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/AcknowledgementController.scala]: At the return from this method we are unnecessary throwing away information we are returning `org.jsoup.nodes.Document`. We may hold this information little longer and convert to `String` when necessary later.

File: app/uk/gov/hmrc/gform/auditing/AuditService.scala (hunk 1 above)
Comment (lutnos) [reply app/uk/gov/hmrc/gform/auditing/AuditService.scala]: Agreed, created https://jira.tools.tax.service.gov.uk/browse/GFC-337

File: app/uk/gov/hmrc/gform/gform/AcknowledgementController.scala (hunk 3 above)
Comment (lutnos) [reply app/uk/gov/hmrc/gform/gform/AcknowledgementController.scala]: Yes.  Further, we should not use jsoup like that at all.  HTML should be generated correctly from views, not generated and then edited.

**Pull request merged**